classes/Member.php

```diff
@@ -376,4 +376,54 @@ public function getEnterLeaveStrings() {
         return $output;
     }
 
+    public static function getRegionalList($postcode, $house, $type) {
+        $db = new \ParlDB;
+
+        $dissolution_dates = array(
+            3 => '2011-03-24',
+            4 => '2011-03-23'
+        );
+        $mreg = array();
+        $constituencies = postcode_to_constituencies($postcode);
+        if ( isset($constituencies[$type]) ) {
+            $cons_name = $constituencies[$type];
+            $q = $db->query("SELECT member.person_id, given_name, family_name, constituency, house
+                FROM member, person_names
+                WHERE
+                member.person_id = person_names.person_id
+                AND constituency = :cons_name
+                AND entered_house >= start_date
+                AND left_house <= end_date
+                AND left_reason = 'still_in_office' AND house = :house",
+                array(
+                    ':house' => $house,
+                    ':cons_name' => $cons_name
+                )
+            );
+            if ( !$q->rows() ) {
+                $q = $db->query("SELECT member.person_id, first_name, last_name, constituency, house
+                    FROM member, person_names
+                    WHERE
+                    member.person_id = person_names.person_id
+                    AND constituency = :cons_name
+                    AND house = :house
+                    AND entered_house >= start_date
+                    AND left_house <= end_date
+                    AND left_house = :dissolution_date",
+                    array(
+                        ':house' => $house,
+                        ':cons_name' => $cons_name,
+                        ':dissolution_date' => $dissolution_dates[$house]
+                    )
+                );
+            }
+
+            for ($i = 0; $i < $q->rows; $i++) {
+                $mreg[] = $q->row($i);
```

Comment:
To match e.g. people.php, if you could call member_full_name here with the right fields, and then just use name rather than given/family in the regional listing templates, to avoid the templates hopefully having to think about names.